QImportP.wsf

```diff
@@ -103,230 +100,551 @@ var QEXE_DEFAULT = Array(
-
-            var lines = WinHttpReq.ResponseText.split("\n");
-            for(var i in lines) {
-                var out = urlhist.toQuicken(lines[i], sym);
-                message("lines i------ " + i + " " + lines[i] + " out=" + out, DEBUG_LEVEL);
-                if (out) {
-                    outFile.WriteLine(out);
-                }
-            }
-        } else {
-            // error, could not retrieve page
-            // some errors may be fine - if start date is recent and no
-            // closing prices exist for that symbol from that date
-            message("Error: could not retreive historical prices for symbol="+sym, ERROR_LEVEL);
-            message("  Code " + WinHttpReq.Status + " retrieving " + url, ERROR_LEVEL);
-        }
-    }
-
-    // get current prices for all symbols, and write to output file
-    // only do this if we did not get any historical dates. This is necessary
-    // because the urls used are different, and the prices may be differently scaled
-    // too (in UK, one url returns pence values, other returns pounds, for example).
-    if (numDays <= 0) {
-        var urlcurr = new UrlCurrForSymbols( symbols );
-        message("URL=" + urlcurr, INFO_LEVEL);
-        WinHttpReq.Open("GET", urlcurr.toUrl(), false);
-        WinHttpReq.Send();
-        if (WinHttpReq.Status == HTTP_STATUS_OK) { // 200
-            message(WinHttpReq.ResponseText, DEBUG_LEVEL);
-            // Was \r\n before April 2015, but that broke, so now just use \n
-            var lines = WinHttpReq.ResponseText.split("\n");
-            for(var i in lines) {
-                var out = urlcurr.toQuicken(lines[i]);
-                message("lines i------ " + i + " " + lines[i] + " out=" + out, DEBUG_LEVEL);
-                if (out) {
-                    outFile.WriteLine(out);
-                }
-            }
-        } else {
-            error = true;
-            message("Error: could not retreive latest prices for symbols="+symbols, ERROR_LEVEL);
-            message("  Code " + WinHttpReq.Status + " retrieving " + urlcurr.toUrl(), ERROR_LEVEL);
-        }
-    }
-
-    // done with file and http
-    outFile.Writeline("^"); 
-    outFile.Close();
-    // WinHttpReq.Close(); // wrong call..?
-    return !error;
-} // getPrices()
+    WinHttpReq.SetTimeouts(30000, 30000, 30000, 30000)
+	
+	// setup domain parsor to process xml data
+	//var myDom = new ActiveXObject("MSXML2.DOMDocument.3.0");
+	//myDom.async=false;
+	//myDom.resolveExternals=false;
+	
+	// setup current and hist url's
+	var urlcurr = new UrlCurr();
+    var urlhist = new UrlHist( numDays );
+
+		
+	// for each symbol, get current then historical price
+	var symbols_array = [];
+	symbols_array = symbols.split(",");
+	for (i in symbols_array) 
+	    { 
+		 var sym = symbols_array[i];
+		 message("Info: Current Price Treatment for symbol=" + sym, INFO_LEVEL);
+
+		// current price for sym
+		var url = urlcurr.toUrl(sym);
+		WinHttpReq.Open("GET", url, false);

```

Comment:
Of course ... in order to debug I have "Microsoft Visual Studio (free version)" installed.
Calling " cscript //D //X  /s:%SYMBOLS%   %ARGUMENTS%   %*" invokes the VS debugger on start where I can set break-points, step through the code and inspect variables at various points. The message displays just give a better idea where to set the break-points